src/dotnet/APIView/APIViewWeb/Repositories/DevopsArtifactRepository.cs

```diff
@@ -50,14 +50,28 @@ public async Task<Stream> DownloadPackageArtifact(string repoName, string buildI
                     downloadUrl = downloadUrl.Split("?")[0] + "?format=" + format + "&subPath=" + filePath;
                 }
                 
-                SetDevopsClientHeaders();
-                var downloadResp = await _devopsClient.GetAsync(downloadUrl);
+                var downloadResp = await GetFromDevopsAsync(downloadUrl);
                 downloadResp.EnsureSuccessStatusCode();
                 return await downloadResp.Content.ReadAsStreamAsync();
             }
             return null;
         }
 
+        private async Task<HttpResponseMessage> GetFromDevopsAsync(string request)
+        {
+            SetDevopsClientHeaders();
+            var downloadResp = await _devopsClient.GetAsync(request);
+            int count = 0;
+            while (downloadResp.StatusCode == HttpStatusCode.TooManyRequests && count < 5)
+            {
+                var retryAfter = downloadResp.Headers.RetryAfter?.ToString() ?? "60";
```

Comment:
I think it would be good to instrument how often we get throttled here via app insights. Being able to track 429 counts from the github api was extremely valuable for diagnosing check enforcer issues around throttling.

Reply:
Done.